client/src/common/types.ts

```diff
@@ -1,3 +1,5 @@
+import { InstanceType } from '../Undo/undo.config';
```

Comment:
This file should define global definitions of types (which are independent from widgets / views types )
Please consider moving InstanceType into this file...